### ISSUE TITLE
Crash report 5.2: PostUploadNotifier.updateNotificationIcon

### ISSUE THREAD

**Issue by maxme** [wordpress-mobile/WordPress-Android#3966]
Only happens on Android 6.0 - looks like a concurrency issue.

```
Fatal Exception: java.lang.RuntimeException: An error occurred while executing doInBackground()
       at android.os.AsyncTask$3.done(AsyncTask.java:309)
       at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
       at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
       at java.util.concurrent.FutureTask.run(FutureTask.java:242)
       at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:234)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
       at java.lang.Thread.run(Thread.java:818)
Caused by java.lang.RuntimeException: bad array lengths
       at android.os.Parcel.readIntArray(Parcel.java:926)
       at android.app.INotificationManager$Stub$Proxy.enqueueNotificationWithTag(INotificationManager.java:777)
       at android.app.NotificationManager.notify(NotificationManager.java:272)
       at android.app.NotificationManager.notify(NotificationManager.java:211)
       at org.wordpress.android.ui.posts.services.PostUploadService$PostUploadNotifier.updateNotificationIcon(PostUploadService.java:893)
       at org.wordpress.android.ui.posts.services.PostUploadService$UploadPostTask.processPostMedia(PostUploadService.java:492)
       at org.wordpress.android.ui.posts.services.PostUploadService$UploadPostTask.doInBackground(PostUploadService.java:235)
       at org.wordpress.android.ui.posts.services.PostUploadService$UploadPostTask.doInBackground(PostUploadService.java:166)
       at android.os.AsyncTask$2.call(AsyncTask.java:295)
       at java.util.concurrent.FutureTask.run(FutureTask.java:237)
       at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:234)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
       at java.lang.Thread.run(Thread.java:818)
```

**Comment by maxme** [comment]
Similar stack trace:

```
Fatal Exception: java.lang.RuntimeException: An error occurred while executing doInBackground()
       at android.os.AsyncTask$3.done(AsyncTask.java:309)
       at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
       at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
       at java.util.concurrent.FutureTask.run(FutureTask.java:242)
       at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:234)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
       at java.lang.Thread.run(Thread.java:818)
Caused by java.lang.RuntimeException: bad array lengths
       at android.os.Parcel.readIntArray(Parcel.java:926)
       at android.app.INotificationManager$Stub$Proxy.enqueueNotificationWithTag(INotificationManager.java:777)
       at android.app.NotificationManager.notify(NotificationManager.java:272)
       at android.app.NotificationManager.notify(NotificationManager.java:211)
       at org.wordpress.android.ui.posts.services.PostUploadService$PostUploadNotifier.updateNotificationMessage(PostUploadService.java:885)
       at org.wordpress.android.ui.posts.services.PostUploadService$UploadPostTask.doInBackground(PostUploadService.java:242)
       at org.wordpress.android.ui.posts.services.PostUploadService$UploadPostTask.doInBackground(PostUploadService.java:166)
       at android.os.AsyncTask$2.call(AsyncTask.java:295)
       at java.util.concurrent.FutureTask.run(FutureTask.java:237)
       at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:234)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
       at java.lang.Thread.run(Thread.java:818)
```

**Comment by maxme** [comment]
Bumped to 5.5 - make sure it's fixed in 5.4.

**Comment by maxme** [comment]
Ref #2858
